indexing-service/src/main/java/io/druid/indexing/worker/WorkerTaskMonitor.java

```diff
@@ -78,8 +85,7 @@ public WorkerTaskMonitor(
     this.cf = cf;
     this.workerCuratorCoordinator = workerCuratorCoordinator;
     this.taskRunner = taskRunner;
-
-    this.exec = Execs.multiThreaded(workerConfig.getCapacity(), "WorkerTaskMonitor-%d");
+    this.exec = Execs.singleThreaded("WorkerTaskMonitor");
```

Comment:
As part of this PR I changed WorkerTaskMonitor to run out of a single thread instead of one thread per task. It made it a lot easier to respond to both status and location updates.